### PR TITLE
step-by-step guide on how to use HWI and Bitcoin Core

### PR DESCRIPTION
**Pull request by 8go**
This page shows in details through screenshots and description on how to use a hardware wallet with Bitcoin Core through the help of HWI. I hope this one-pager helps people to see how it works, takes their fear and increments the use of hardware wallets with Bitcoin Core.

### CODE REVIEW

File: docs/walk-through/README.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+TREZORFINGERPRINT_TESTNET="yourTrezorFingerprint" # shown by "hwi enumerate"
+# in this example we use SEGWIT BECH32 ADDRESSES
+DERIVATIONPATH_TESTNET=1 # testnet uses derivation paths like m/84h/1h/0h/0/* and m/84h/1h/0h/1/* on testnet
+DERIVATIONPATH_MAINNET=0 # mainnet uses derivation paths like m/84h/0h/0h/0/* and m/84h/0h/0h/1/* on testnet
+# if the mainnet path is used on testnet, it will work too but Trezor device gives warnings 
+# of unknown address on Trezor display
+wallet=wallet.test
+rec=$(hwi --testnet -f $TREZORFINGERPRINT_TESTNET getkeypool --wpkh --path m/84h/${DERIVATIONPATH_TESTNET}h/0h/0/* --keypool 0 1000)
+chg=$(hwi --testnet -f $TREZORFINGERPRINT_TESTNET getkeypool --wpkh --path m/84h/${DERIVATIONPATH_TESTNET}h/0h/1/* --keypool --internal 0 1000)
+while ! bitcoin-cli -testnet -rpcuser=test -rpcpassword=somedifficultpassword -rpcclienttimeout=0 createwallet "$wallet" true ; do sleep 60; done
```
Comment (achow101) [inline docs/walk-through/README.md:32]: It's unnecessary to specify `-rpcuser`, `-rpassword`, and `-rpcclienttimeout`. If you are specifying the datadir (as you did in your `bitcoind` command), then specifying `-datadir` for `bitcoin-cli` is sufficient.

I would suggest that for this guide, we avoid specifying addiional command line options for `bitcoind` and `bitcoin-cli`.

File: docs/walk-through/README.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+hwi.py enumerate # this shows you the fingerprint of hardware wallet
+TREZORFINGERPRINT_TESTNET="yourTrezorFingerprint" # shown by "hwi enumerate"
+# in this example we use SEGWIT BECH32 ADDRESSES
+DERIVATIONPATH_TESTNET=1 # testnet uses derivation paths like m/84h/1h/0h/0/* and m/84h/1h/0h/1/* on testnet
+DERIVATIONPATH_MAINNET=0 # mainnet uses derivation paths like m/84h/0h/0h/0/* and m/84h/0h/0h/1/* on testnet
+# if the mainnet path is used on testnet, it will work too but Trezor device gives warnings 
+# of unknown address on Trezor display
+wallet=wallet.test
+rec=$(hwi --testnet -f $TREZORFINGERPRINT_TESTNET getkeypool --wpkh --path m/84h/${DERIVATIONPATH_TESTNET}h/0h/0/* --keypool 0 1000)
+chg=$(hwi --testnet -f $TREZORFINGERPRINT_TESTNET getkeypool --wpkh --path m/84h/${DERIVATIONPATH_TESTNET}h/0h/1/* --keypool --internal 0 1000)
```
Comment (achow101) [inline docs/walk-through/README.md:31]: `--wpkh` was replaced with `--addr-type wpkh`.

Additionally, it is unnecessary to specify `--path` if you are planning on just using the default path. To get the testnet path, you can use the `--testnet` option.

Lastly, without `--path`, `getkeypool` will return both change and receiving descriptors, so it is unnecessary to have call it twice.

File: docs/walk-through/README.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+while ! bitcoin-cli -testnet -rpcuser=test -rpcpassword=somedifficultpassword -rpcclienttimeout=0 createwallet "$wallet" true ; do sleep 60; done
+while ! bitcoin-cli -testnet -rpcuser=test -rpcpassword=somedifficultpassword -rpcclienttimeout=0 -rpcwallet="$wallet" importmulti "$rec" ; do sleep 60; done
```
Comment (achow101) [inline docs/walk-through/README.md:33]: Having these commands be in a `while` loop is unnecessary and could have unintended side effects (re-imports may have some issues).

**Review by achow101 — COMMENTED** [review]
Concept ACK

It would be nice for the screenshots to have more descriptive names rather than just the date and time they were made.

Additionally, since we are moving to using Sphinx and RST for docs soon (see #464), it would be nice for this to be formatted with RST so that it will work without changes in Sphinx.

**Comment by 8go** [comment]
Thanks Andrew @achow101 for your valuable feedback on PR #413.
I addressed nearly all of your issues and followed your suggestions.
Your suggestions have been added via the 3 commits 067e0c06f473d4fdd0519597b308877d0c4637d0 , 58379f6652531fb4b66c7be4997f72f7ef2a5c85 and 30b1bb4bb600df9fc79631410813c0162bd229cc.

Please have a look at the improved version. Let me know if there is anything else that needs to be done.

Thanks again and cheers!

File: docs/walk-through/README.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
+# "Step-by-step Walk-Through" or 
+# How to Use a Hardware Wallet with Bitcoin Core Wallet
+
+Summary: On this page we describe step-by-step and show in screenshots how to use a hardware wallet and HWI 
+together with a Bitcoin Core Wallet. As hardware wallet example we have used a Trezor.
+
+## Create a watch-only Bitcoin Core wallet for Trezor
+Create your watch-only Bitcoin Core Wallet as well described in [Using Bitcoin Core with Hardware Wallets](../bitcoin-core-usage.md). 
+You find all the details in this link. But in summary, one opens one terminal and runs `bitdoind`. E.g.
```
[the remark this answers is not in the document]
Comment (8go) [reply docs/walk-through/README.md]: This was not the only typo. As a matter of fact upon reading it again a few more times, a found several more typos. Fixed in the new commits.

File: docs/walk-through/README.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+```
+bitcoind -testnet  -datadir=$HOME/.bitcoin-testnet -rpcallowip=127.0.0.1 -rpcuser=test -rpcpassword=somedifficultpassword
```
[the remark this answers is not in the document]
Comment (8go) [reply docs/walk-through/README.md]: I agree. I took these parameters out in all places. 

Fixed in the new commits.

File: docs/walk-through/README.md (hunk 1 above)
Comment (8go) [reply docs/walk-through/README.md]: OK, I removed the -rpcuser, -rpcpassword options. Ok, I will take out the -rpcclienttimeout option as well in all places.
Done in commit 8e6de745d625d5ebc481f0ccddf15e3dba91da3b.

File: docs/walk-through/README.md (hunk 2 above)
Comment (8go) [reply docs/walk-through/README.md]: Ok, Andrew, I replaced the `--wpkh` with `--addr-type wpkh` so that this option is up-to-date.

I know and I understand that by removing `--path` that with a single command both receiving and change address will be handled at once and that there are fewer calls, i.e. the script is shorter. However, I copied this code out of my script that I used for testing many different scenarios. I find it definitely helpful, for users to see which address they are creating, so the next time around e.g. they can do address 1000 through 2000, etc. I personally also found it useful to see path so that the script can be adapted to different networks. In my humble opinion it helps the user understand better and through these examples the user learns more and can adapt this script to more situations. 

In short, I was not going for the shortest code, or the simplest solution, but for the solution that teaches the most and demos the most info. As such, I have left the options `--path` and the additional `--keypool` arguments in place. 

What do you think @achow101 ? Reasonable?

File: docs/walk-through/README.md (hunk 3 above)
Comment (8go) [reply docs/walk-through/README.md]: Again, this was taken from a real world script. When doing tests with failing networks, inkecting other faults, etc. the `while` loop is useful. It checks the return code anyway to avoid duplicating the imports. 

Anyway this might not be needed for the average user. I followed your advice and removed the while loops as suggested.

**Comment by 8go** [comment]
Hi Andrew,

All done. 
The commits are squashed.
Screenshots have nice descriptive names now. 

Please have a look at this revised version. Thanks!

**Comment by achow101** [comment]
ACK 7de0237c5590efde7364209c30fc3dbb77544786

**Comment by 8go** [comment]
Great! The page is live at
https://hwi.readthedocs.io/en/latest/examples/walkthrough/walkthrough.html
